union.go

```diff
@@ -0,0 +1,58 @@
+package rwvfs
+
+import (
+	"io"
+	pathpkg "path"
+
+	"golang.org/x/tools/godoc/vfs"
+)
+
+// Union returns a new FileSystem which is the union of the provided file systems.
+func Union(fileSystems ...FileSystem) FileSystem {
+	if len(fileSystems) == 0 {
+		return Map(nil)
+	}
+	if len(fileSystems) == 1 {
+		return fileSystems[0]
+	}
+
+	ns := vfs.NameSpace{}
+	for _, fs := range fileSystems {
+		ns.Bind("/", fs, "/", vfs.BindAfter)
+	}
+	return &unionFs{
+		NameSpace:   ns,
+		fileSystems: fileSystems,
+	}
+}
+
+type unionFs struct {
+	vfs.NameSpace
+	fileSystems []FileSystem
+}
+
+func (fs *unionFs) resolve(path string) FileSystem {
+	parent := pathpkg.Dir(path)
+	for _, fs2 := range fs.fileSystems {
+		if _, err := fs2.Stat(parent); err == nil {
```

Comment:
What do you think of the following?
```go
func (fs *unionFs) resolve(path string) FileSystem {
	parent := pathpkg.Dir(path)
	for _, fs2 := range fs.fileSystems {
		if _, err := fs2.Stat(parent); err != nil && os.IsNotExist(err) {
			continue;
		}
		return fs2
	}
	return fs.fileSystems[0]
}
```
The reason for not returning an error here is that I want to return an error for `path` itself, not for its parent directory.